sandbox/libs/authn/docs/openapi.yaml

```diff
@@ -0,0 +1,79 @@
+openapi: 3.0.0
+info:
+  version: 1.0.0
+  title: OpenSearch Identity Permission API
+  license:
+    name: Apache 2.0
+jsonSchemaDialect:
+paths:
+  cluster/permissions:
+    get:
+      requestBody:
+        description: Get a list of all permissions granted to a given principal
+        content:
+          application/json:
+            name: username
+            description: the username of the principal name to get the permissions for
+            required: true
+            schema:
+              type: string
+              format: utf-8
+      responses:
+        "200":
+          description: Return a 200 status to indicate success and show the requested permissions
+          application/json:
+              description: A list of permission strings
+        default:
+          description: illegal argument exception
+    put:
+      requestBody:
+        description: Add a permission
+        content:
+          application/json:
+            - name: principal name
+              description: the principal name to grant a permission to
+              required: true
+              schema:
+                type: string
+                format: utf-8
+            - name: permission string
+              description: the permission string for the permission being granted
+              required: true
+              schema:
+                type: string
+                format: utf-8
+      responses:
+        "200":
+          description: Return a 200 status to indicate that the permission was added
+          application/json:
+            schema:
+              type: string
+              format: utf-8
+        default:
+          description: unexpected error
+    delete:
+      requestBody:
+        description: Delete a permission
+        content:
+          application/json:
+            - name: username
+              description: the username of the principal to delete a permission from
+              required: true
+              schema:
+                type: string
+                format: utf-8
+            - name: permission string
```

Comment:
Do you mean in general or just in the yaml? If in general, I can make that switch though the idea had been to be as descriptive with the fields as possible. I was not sure if it would be confusing to call the field a permission when a permission is its own object separate from the string that represents it. 

If you just mean in the documentation, I think you are supposed to put the variable name in the "name" fields which is why this is here. Just let me know what you meant.

Reply:
In general. IMO conciseness == easier to read and maintain. 

`Grant <permission> to <principal>`

is more concise then

`Grant <permissionString> to <principalString>`

Reply:
I see, you're definitely right when you put it like that.